### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by jessiemongeon1**
- Add community sample projects and resources
- Add more error codes
- Add specifying specific dfx version and passing args to moc in dfx.json

### CODE REVIEW

File: doc/md/getting-started/dev-env.md [hunk 1]
```diff
@@ -111,3 +113,32 @@ For example, to use base library version `0.9.0`, use the command:
...
+
+Here is an example `dfx.json` canister configuration that uses `args`:
+
+```json
+...
+  "canisters": {
+    "hello": {
+      "type": "motoko",
+      "main": "src/hello/main.mo",
+      "args": "-v --compacting-gc -no-check-ir"
```
Comment (luc-blaeser) [inline doc/md/getting-started/dev-env.md:140]:
```suggestion
      "args": "-v --incremental-gc"
```

Comment (luc-blaeser) [reply doc/md/getting-started/dev-env.md:140]: Maybe, we could encourage to use the new GC :).
I am not sure whether `-no-check-ir` is really useful/visible for the users.

File: doc/md/getting-started/examples.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+- [Awesome Motoko](https://github.com/ic123-xyz/awesome-motoko): A list of Motoko projects, tools, and resources.
+
+- [DFINITY Examples](https://github.com/dfinity/examples/tree/master/motoko): Simple Motoko sample projects and templates.
+
+- [ICP developer journey](https://internetcomputer.org/docs/current/tutorials/developer-journey/): A 36-part series focused on Motoko development for ICP canisters.
+
+- [Motoko book](https://github.com/Web3NL/motoko-book): An educational book on Motoko.
+
+- [Motoko Bootcamp](https://github.com/motoko-bootcamp/education): Lessons and lectures on Motoko development.
```
Comment (luc-blaeser) [inline doc/md/getting-started/examples.md:108]: Not sure whether we should perhaps add https://github.com/luc-blaeser/auction, the Motoko CySep summer school workshop at KTH Sweden.

Comment (luc-blaeser) [reply doc/md/getting-started/examples.md:108]: Very nice collection!

**Review by luc-blaeser — APPROVED** [review]
Very nice! Thank you for this PR.

File: doc/md/getting-started/examples.md (hunk 2 above)
Comment (jessiemongeon1) [reply doc/md/getting-started/examples.md:110]: Added the auction example :)

**Pull request merged**